scripts/check_python.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import argparse
+import multiprocessing
+import os
+import subprocess
+import sys
+from concurrent import futures
+from pathlib import Path
+from typing import List
+
+
+def run_cmd(cmd, verbose):
+    if verbose:
+        print(f"Running: {cmd}")
+    completed = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False, encoding="UTF-8")
+    return completed
+
+
+if __name__ == "__main__":
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description="Check or fix Python like code.")
+    parser.add_argument(
+        "--fix",
+        "-fix",
+        "-f",
+        default=False,
+        action="store_true",
+        help="Fix the sources instead of just checking their formatting.",
+    )
+    parser.add_argument(
+        "--verbose", "-v", default=False, action="store_true", help="Verbosely print what I am doing.",
+    )
+    parser.add_argument(
+        "--pylint", default=False, action="store_true", help="Run pylint and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--black", default=False, action="store_true", help="Run black and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--isort", default=False, action="store_true", help="Run isort and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--exclude",
+        default=[],
+        action="append",
+        help="Add a pattern to exclude from processing. The pattern can be a path or a glob.",
+    )
+    parser.add_argument("roots", type=str, nargs="+", help="The root paths to search from, or files to check.")
+    args = parser.parse_args()
+
+    if not args.pylint and not args.black and not args.isort:
+        args.pylint = True
+        args.black = True
+        args.isort = True
+
+    exclude_paths = set(args.exclude)
+    # For backwards compat check an old environment variable.
+    exclude_paths.update(os.environ.get("PRUNE_PATHS", "").split(":"))
```

Comment:
Might be worth a quick check if anyone actually depends on `PRUNE_PATHS` and I'm sure they can learn to pass their arguments on the command line instead.

Reply:
I'm gonna merge as is. If there was no response to my question on code related in a couple days I will create another PR to remove PRUNE_PATHS entirely.